tsdb/engine/tsm1/engine.go

```diff
@@ -338,6 +362,11 @@ func (e *Engine) SetLogOutput(w io.Writer) {
 func (e *Engine) LoadMetadataIndex(shardID uint64, index *tsdb.DatabaseIndex) error {
 	now := time.Now()
 
+	// Open the index if it's not already open.
+	if err := index.Open(); err != nil {
```

Comment:
Though now I see that it should be `index Index` not `index *tsdb.DatabaseIndex`...